checkQC/qc_engine.py

```diff
@@ -100,9 +102,14 @@ def _run_parsers(self):
     def _compile_reports(self):
         reports = {"exit_status": 0}
         for handler in self._handlers:
-            handler_report = handler.report()
+            handler_name = type(handler).__name__
```

Comment:
This is the code I'm thinking that we can get rid of if we follow the approach I suggest above. A neat thing about that solution is that it preserves the separation of concerns between the layers here. I.e. the handler alone is responsible for figuring out if something should be an error or a warning, and the engine can just parse it as before.